### PR TITLE
feat(apprunner): make `Service` implement `IGrantable`

### LINKED ISSUE #26089: apprunner: Allow `instanceRole` to be obtained from `Service` instances.
### Describe the feature

Currently, `instanceRole` can only be set from the constructor.
If it is possible to retrieve it even after initialization like [`obtainExecutionRole`](https://github.com/aws/aws-cdk/blob/v2.85.0/packages/aws-cdk-lib/aws-ecs/lib/base/task-definition.ts#L704) in `TaskDefinition` of ECS, it would be useful to be able to grant privileges with the `grant` method.

### Use Case

Use as follows:

```typescript
declare const bucket: IBucket;

const service = new apprunner.Service(this, 'Service', {
    source: apprunner.Source.fromEcrPublic({
        imageConfiguration: { port: 8000 },
        imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
    }),
});

bucket.grantRead(service.obtainInstanceRole());
```

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.85.0

### Environment details (OS name and version, etc.)

N/A

### PR DESCRIPTION
**Pull request by lpizzinidev**
Implementing `IGrantable` for cases when it's needed to grant permissions to a `Service` instance.

For example:
```
declare const bucket: IBucket;

const service = new apprunner.Service(this, 'Service', {
    source: apprunner.Source.fromEcrPublic({
        imageConfiguration: { port: 8000 },
        imageIdentifier: 'public.ecr.aws/aws-containers/hello-app-runner:latest',
    }),
});

bucket.grantRead(service);
```

Closes #26089.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts [hunk 1]
```diff
@@ -1140,6 +1140,17 @@ export class Service extends cdk.Resource {
     this.secrets.push({ name: name, value: secret.arn });
   }
 
+  /**
+   * This method exposes the Instance Role after creating it if not set.
+   * @returns iam.IRole
+   */
+  public obtainInstanceRole(): iam.IRole {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts:1147]: What if we instead made `Service` implement `IGrantable`? Then we could just do
things like

```ts
bucket.grantRead(service);
```

**Review by corymhall — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts [hunk 2]
```diff
@@ -1052,6 +1053,7 @@ export class Service extends cdk.Resource {
     this.props = props;
 
     this.instanceRole = this.props.instanceRole;
+    this.grantPrincipal = this.instanceRole || new iam.UnknownPrincipal({ resource: this });
```
Comment (corymhall) [inline packages/@aws-cdk/aws-apprunner-alpha/lib/service.ts:1056]: I think we should just go ahead and create the role if the user doesn't provide it. I'm not sure why we try to only generate one if needed.

**Review by corymhall — CHANGES_REQUESTED** [review]
Can you also add an `addToRolePolicy` method? I'm also going to change this to a `feat`. Adding the auto creation of the role should cause the integ test to update and we'll need a README update.

**Comment by lpizzinidev** [comment]
Exemption request.
I've updated the related integration tests.

**Review by mrgrain — APPROVED** [review]

**Comment by lpizzinidev** [comment]
> Can you also add an `addToRolePolicy` method? I'm also going to change this to a `feat`. Adding the auto creation of the role should cause the integ test to update and we'll need a README update.

@corymhall I've implemented the requested changes.

**Review by corymhall — APPROVED** [review]

**Pull request merged**